components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -71,4 +80,47 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
         }
         return null;
     }
+
+    public Response updateRolesForScope(ScopeListDTO body, MessageContext messageContext) throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(RestApiUtil.getLoggedInUsername());
+        //read from tenant-conf.json
+        JsonObject existingTenantConfObject = new JsonObject();
+        try {
+            APIMRegistryService apimRegistryService = new APIMRegistryServiceImpl();
+            String existingTenantConf = apimRegistryService.getConfigRegistryResourceContent(tenantDomain,
+                    APIConstants.API_TENANT_CONF_LOCATION);
+             existingTenantConfObject = new JsonParser().parse(existingTenantConf).getAsJsonObject();
+        } catch (RegistryException | UserStoreException e) {
+            APIUtil.handleException("Couldn't read tenant configuration from tenant registry", e);
+        }
+        try {
+
+            JSONObject responseJson = new JSONObject();
+            JSONArray scopeJson = new JSONArray();
+
+            for(int i = 0; i < body.getList().size(); i++) {
+                JSONObject scopeRoleJson = new JSONObject();
+                scopeRoleJson.put("Name", body.getList().get(i).getName());
+                scopeRoleJson.put("Roles", body.getList().get(i).getRoles());
+                scopeJson.put(scopeRoleJson);
+            }
+
+            responseJson.put("Scope", scopeJson);
+            //Here we are removing RESTAPIScopes from the existing tenant-conf
+            // Adding new RESTAPIScopes to the existing tenant-conf.
+            existingTenantConfObject.remove("RESTAPIScopes");
+            Gson gson = new Gson();
+            existingTenantConfObject.add("RESTAPIScopes",
+                    gson.fromJson(responseJson.toString(), JsonElement.class));
+            APIUtil.updateTenantConf(existingTenantConfObject.toString(),tenantDomain);
+            Map<String, String> scopeRoleMapping = APIUtil.getRESTAPIScopesForTenant(MultitenantUtils
+                    .getTenantDomain(RestApiUtil.getLoggedInUsername()));
+            ScopeListDTO scopeListDTO = SystemScopesMappingUtil.fromScopeListToScopeListDTO(scopeRoleMapping);
+            return Response.ok().entity(scopeListDTO).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Error when updating the list of scopes-role mapping.";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
```

Comment:
We don't need to catch APIManagementException from the REST API layer. Throw it from the method

Reply:
Fixed with bb70f77